docs/troubleshooting.md

```diff
@@ -214,10 +209,10 @@ RUN chmod +x /usr/local/bin/dumb-init
 # ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD true
 
 # Install puppeteer so it's available in the container.
-RUN npm i puppeteer
-
+RUN npm i puppeteer \
 # Add user so we don't need --no-sandbox.
-RUN groupadd -r pptruser && useradd -r -g pptruser -G audio,video pptruser \
+# same layer as npm install to keep re-chowned files from using up several hundred MBs more space
```

Comment:
I'm surprised this change has any effect on the resulting image. Can you please explain why? (or maybe you point me towards some documentation regarding this..)

Reply:
https://medium.com/@jessgreb01/digging-into-docker-layers-c22f948ed612 is good for explaining layers.

Reply:
@neilsh what's the savings in image size with these changes?

Reply:
The main culprit is the `chown -R pptruser:pptruser /node_modules` line. If  it's on a separate docker operation from the `npm install` it essentially creates a completely separate copy of the chrome binary in the node_modules. Just that `chown` layer adds an extra 323MB to the image. https://medium.com/@lmakarov/the-backlash-of-chmod-chown-mv-in-your-dockerfile-f12fe08c0b55

@ebidel overall image size change I saw was 1.21GB -> 868MB